openassessment/xblock/test/test_openassessment.py

```diff
@@ -91,95 +96,246 @@ def test_student_view_workflow_error(self, xblock):
+    @ddt.unpack
+    def test_load_student_view_with_dates(self, time_zone, expected_date):
         """OA XBlock returns some HTML to the user.
 
         View basic test for verifying we're returned some HTML about the
         Open Assessment XBlock. We don't want to match too heavily against the
         contents.
         """
-        xblock_fragment = self.runtime.render(xblock, "student_view")
-        self.assertTrue(xblock_fragment.body_html().find("Openassessmentblock"))
-
-        # Validate Submission Rendering.
-        submission_response = xblock.render_submission({})
-        self.assertIsNotNone(submission_response)
-        self.assertTrue(submission_response.body.find("openassessment__response"))
-        self.assertTrue(submission_response.body.find("April"))
-
-    @scenario('data/basic_scenario.xml')
-    def test_formatted_dates(self, xblock):
-
-        # Set start/due dates
-        xblock.start = dt.datetime(2014, 4, 1, 1, 1, 1)
-        xblock.due = dt.datetime(2014, 5, 1)
-
-        request = namedtuple('Request', 'params')
-        request.params = {}
-        resp = xblock.render_peer_assessment(request)
-        self.assertTrue(resp.body.find('Tuesday, April 01, 2014'))
-        self.assertTrue(resp.body.find('Thursday, May 01, 2014'))
-
-    @scenario('data/basic_scenario.xml')
-    def test_formatted_dates_for_beta_tester_with_days_early(self, xblock):
-        """Test dates for beta tester with days early"""
-
-        # Set start/due dates
-        xblock.start = dt.datetime(2014, 4, 6, 1, 1, 1)
-        xblock.due = dt.datetime(2014, 5, 1)
-        xblock.xmodule_runtime = Mock(
-            course_id='test_course',
-            anonymous_student_id='test_student',
-            days_early_for_beta=5,
-            user_is_staff=False,
-            user_is_beta_tester=True
-        )
-        self.assertEqual(xblock.xmodule_runtime.days_early_for_beta, 5)
-        request = namedtuple('Request', 'params')
-        request.params = {}
-        resp = xblock.render_peer_assessment(request)
-        self.assertTrue(resp.body.find('Tuesday, April 01, 2014'))
-        self.assertTrue(resp.body.find('Thursday, May 01, 2014'))
+        with patch('openassessment.xblock.submission_mixin.get_current_time_zone') as time_zone_fn:
+            time_zone_fn.return_value = pytz.timezone(time_zone)
 
+            xblock = self.load_scenario('data/dates_scenario.xml')
+            xblock_fragment = self.runtime.render(xblock, "student_view")
+            self.assertIn("OpenAssessmentBlock", xblock_fragment.body_html())
+
+            # Validate Submission Rendering.
+            submission_response = xblock.render_submission({})
+            self.assertIsNotNone(submission_response)
+            self.assertIn("openassessment__response", submission_response.body)
+            self.assertIn(expected_date, submission_response.body)
+
+    time_zone_fn_path = 'openassessment.xblock.peer_assessment_mixin.get_current_time_zone'
```

Comment:
It's weird to see this definition here. can you move it to the top of the class, and ALL_CAPS to indicate a constant?